### PR TITLE
PE-838 ESM Update, Lerp removal, sb increase

### PR DESCRIPTION
**Pull request by brianmcmichael**
# Description

# Contribution Checklist

- [x] PR title starts with `(PE-<TICKET_NUMBER>)`
- [x] Code approved
- [x] Tests approved
- [x] CI Tests pass

# Checklist

- [x] Every contract variable/method declared as public/external private/internal
- [x] Consider if this PR needs the `officeHours` modifier override
- [x] Verify expiration (`30 days` unless otherwise specified)
- [x] Verify hash in the description matches [here](https://emn178.github.io/online-tools/keccak_256.html)
- [x] Validate all addresses used are in Goerli changelog or known
- [x] Notify any external teams affected by the spell so they have the opportunity to review
- [x] Deploy spell to Goerli `ETH_GAS="XXX" ETH_GAS_PRICE="YYY" make deploy`
- [x] Ensure contract is verified on `Goerli` etherscan
- [x] Change test to use Goerli spell address and deploy timestamp
- [x] Run `make archive-spell` or `make date="YYYY-MM-DD" archive-spell` to make an archive directory and copy `Goerli-DssSpell.sol`, `Goerli-DssSpell.t.sol`, `Goerli-DssSpell.t.base.sol`, and `Goerli-DssSpellCollateralOnboarding.sol`
- [ ] `squash and merge` this PR

### CODE REVIEW

**Comment by talbaneth** [comment]
Verified the list of contracts in the spell matches Julien's (and mine).

**Comment by talbaneth** [comment]
@brianmcmichael, can add the relevant poll and exec links:
https://vote.makerdao.com/polling/QmQSVmrh?network=mainnet#poll-detail
https://vote.makerdao.com/executive/template-executive-vote-temporarily-prevent-surplus-flap-auctions-and-mkr-burn-january-24-2022?network=mainnet#proposal-detail

**Comment by talbaneth** [comment]
also - should we change the ESM address in test/addresses_goerli.sol? (addr["MCD_ESM"])

**Comment by brianmcmichael** [comment]
> also - should we change the ESM address in test/addresses_goerli.sol? (addr["MCD_ESM"])

I updated this and added a check for the expected threshold to the system tests. You'll want to be sure to pull these updates over to the mainnet spell

File: src/Goerli-DssSpell.t.base.sol [hunk 1]
```diff
@@ -220,13 +220,14 @@ contract GoerliDssSpellTestBase is Config, DSTest, DSMath {
...
-            vow_hump_min:          60 * MILLION,            // In whole Dai units
-            vow_hump_max:          90 * MILLION,            // In whole Dai units
+            vow_hump_min:          250 * MILLION,           // In whole Dai units
+            vow_hump_max:          250 * MILLION,           // In whole Dai units
             flap_beg:              400,                     // in basis points
             flap_ttl:              30 minutes,              // in seconds
             flap_tau:              72 hours,                // in seconds
             cat_box:               20 * MILLION,            // In whole Dai units
             dog_Hole:              100 * MILLION,           // In whole Dai units
+            esm_min:               100 * THOUSAND,           // In whole MKR units
```
Comment (talbaneth) [inline src/Goerli-DssSpell.t.base.sol:230]:
```suggestion
            esm_min:               100 * THOUSAND,          // In whole MKR units
```

File: src/Goerli-DssSpell.sol [hunk 2]
```diff
@@ -42,64 +42,255 @@ contract DssSpellAction is DssAction, DssSpellCollateralOnboardingAction {
     //    https://ipfs.io/ipfs/QmefQMseb3AiTapiAKKexdKHig8wroKuZbmLtPLv4u2YwW
     //
 
-    // --- Rates ---
-    uint256 constant ZERO_FIVE_PCT_RATE      = 1000000000158153903837946257;
-    uint256 constant TWO_FIVE_PCT_RATE       = 1000000000782997609082909351;
-    uint256 constant THREE_PCT_RATE          = 1000000000937303470807876289;
+    address constant SB_LERP = address(0x98bbDDe76EB5753578280bEA86Ed8401f2831213);
+    address constant NEW_MCD_ESM = address(0x023A960cb9BE7eDE35B433256f4AfE9013334b55);
+    bytes32 constant MCD_ESM = "MCD_ESM";
```
Comment (talbaneth) [inline src/Goerli-DssSpell.sol:47]: Don't think this adds any value, would prefer to just use "MCD_ESM" inline.

**Comment by talbaneth** [comment]
See 2 neat suggestions.
Also added a PR with 2 more tests (ported from previoues ESM replacement).
After those we are good to deploy imo.

File: src/Goerli-DssSpell.sol (hunk 2 above)
Comment (brianmcmichael) [reply src/Goerli-DssSpell.sol:48]: So strictly speaking we should be limiting the number of times we print out tags to reduce the likelihood of homograph errors/attacks.

For example:

```
bytes32("MCD_ESM")
0x4d43445f45534d000000000000000000000000000000000000000000000000

bytes32("МCD_ESM")
0xd09c43445f45534d0000000000000000000000000000000000000000000000
```

If we use a tag once we can inline it, but for multiple uses of a tag we probably just want to use it once so we ensure that all instances use the same bytes32 string.

Comment (talbaneth) [reply src/Goerli-DssSpell.sol:48]: Sure not a big deal either way.

**Comment by talbaneth** [comment]
<pre>dapp-test: rpc block: latest
Running 16 tests for src/Goerli-DssSpell.t.sol:DssSpellTest
<font color="#4E9A06">[PASS]</font> testFail_notScheduled() (gas: 7161)
<font color="#4E9A06">[PASS]</font> test_auth_in_sources() (gas: 4625876)
<font color="#4E9A06">[PASS]</font> test_use_eta() (gas: 331363)
<font color="#4E9A06">[PASS]</font> testOnTime() (gas: 2124234)
<font color="#4E9A06">[PASS]</font> test_nextCastTime() (gas: 332214)
<font color="#4E9A06">[PASS]</font> testFailTooEarly() (gas: 3573)
<font color="#4E9A06">[PASS]</font> test_bytecode_matches() (gas: 5571566)
<font color="#4E9A06">[PASS]</font> testFailFireESM() (gas: 2244571)
<font color="#4E9A06">[PASS]</font> test_auth() (gas: 3953571)
<font color="#4E9A06">[PASS]</font> testSpellIsCast_GENERAL() (gas: 13453749)
<font color="#4E9A06">[PASS]</font> testFailWrongDay() (gas: 3528)
<font color="#4E9A06">[PASS]</font> testCastCost() (gas: 2125713)
<font color="#4E9A06">[PASS]</font> testLerps() (gas: 2147387)
<font color="#4E9A06">[PASS]</font> testESMOffboarding() (gas: 6780159)
<font color="#4E9A06">[PASS]</font> testFireESM() (gas: 2453331)
<font color="#4E9A06">[PASS]</font> testFailTooLate() (gas: 3549)
</pre>

**Review by talbaneth — APPROVED** [review]
LGTM for deployment

**Review by godsflaw — APPROVED** [review]
LGTM

```haskell
Running 16 tests for src/Goerli-DssSpell.t.sol:DssSpellTest
[PASS] testFail_notScheduled() (gas: 7161)
[PASS] test_auth_in_sources() (gas: 4625876)
[PASS] test_use_eta() (gas: 331363)
[PASS] testOnTime() (gas: 2124234)
[PASS] test_nextCastTime() (gas: 332214)
[PASS] testFailTooEarly() (gas: 3573)
[PASS] test_bytecode_matches() (gas: 5571566)
[PASS] testFailFireESM() (gas: 2244571)
[PASS] test_auth() (gas: 3953571)
[PASS] testSpellIsCast_GENERAL() (gas: 13453749)
[PASS] testFailWrongDay() (gas: 3528)
[PASS] testCastCost() (gas: 2125713)
[PASS] testLerps() (gas: 2147387)
[PASS] testESMOffboarding() (gas: 6780159)
[PASS] testFireESM() (gas: 2453331)
[PASS] testFailTooLate() (gas: 3549)
```

**Review by godsflaw — APPROVED** [review]

**Pull request merged**